quil-py/src/program/source_map.rs

```diff
@@ -0,0 +1,147 @@
+use pyo3::{types::PyInt, Py};
+use quil_rs::program::{
+    CalibrationExpansion, CalibrationSource, MaybeCalibrationExpansion,
+    ProgramCalibrationExpansion, ProgramCalibrationExpansionSourceMap, SourceMap, SourceMapEntry,
+};
+use rigetti_pyo3::{impl_repr, py_wrap_type, py_wrap_union_enum, pyo3::pymethods, PyWrapper};
+
+use crate::instruction::{PyCalibrationIdentifier, PyMeasureCalibrationIdentifier};
+
+use super::PyProgram;
+
+type CalibrationExpansionSourceMap = SourceMap<usize, CalibrationExpansion>;
+type CalibrationExpansionSourceMapEntry = SourceMapEntry<usize, CalibrationExpansion>;
+type ProgramCalibrationExpansionSourceMapEntry = SourceMapEntry<usize, MaybeCalibrationExpansion>;
+
+py_wrap_type! {
+    #[derive(Debug, PartialEq)]
+    PyCalibrationExpansion(CalibrationExpansion) as "CalibrationExpansion"
+}
+
+impl_repr!(PyCalibrationExpansion);
+
+#[pymethods]
+impl PyCalibrationExpansion {
+    pub fn calibration_used(&self) -> PyCalibrationSource {
+        self.as_inner().calibration_used().into()
+    }
+
+    // Reviewer: is there a better return type?
+    pub fn range(&self) -> (usize, usize) {
+        (self.as_inner().range().start, self.as_inner().range().end)
+    }
+
+    pub fn expansions(&self) -> PyCalibrationExpansionSourceMap {
+        self.as_inner().expansions().into()
+    }
+}
+
+py_wrap_type! {
+    #[derive(Debug, PartialEq)]
```

Comment:
Any Python type that can derive PartialEq should also use the `impl_eq!` macro so that Python can leverage it as well.

Reply:
good call - added some, will sweep for rest on next pass